examples/mnit_ballz/src/ballz_android.nit

```diff
@@ -38,32 +44,65 @@ redef class App
 		proximity.enabled = true
 		maximum_fps = 50
 
-		super
-	end
-
-	redef fun on_create
-	do
-		super
-		screen = new Screen(self, display.as(Display))
+		game = new Game(display.width.to_f, display.height.to_f)
 	end
 
 	redef fun frame_core(display)
 	do
-		var screen = self.screen
-		if screen != null then
-			screen.game.do_turn
-			screen.do_frame(display)
+		var game = game
+		if game != null then
+			game.do_turn
+			game.draw(display, assets)
 		end
 	end
 
 	redef fun input(ie)
-	do	
-		if ie isa QuitEvent then 
+	do
+		if paused then return false
+		if ie isa QuitEvent then
 			quit = true
 			return true
 		end
-		if screen != null then
-			return screen.input(ie)
+		var game = game
+		if game != null then
+			return game.input(ie)
+		end
+		return false
+	end
+end
+
+redef class Ball
+
+	redef fun intercepts(event)
+	do
+		if event isa ASensorAccelerometer then
+			acceleration(event.x, event.y)
+		else if event isa ASensorMagneticField then
```

Comment:
Yes, and this code may also be useful for next pull requests if other events are used